firedrake/functionspaceimpl.py

```diff
@@ -378,17 +378,23 @@ def make_function_space(cls, mesh, element, name=None):
             new = cls.create(new, mesh)
         return new
 
-    def reconstruct(self, mesh=None, name=None, **kwargs):
+    def reconstruct(self, mesh=None, element=None, boundary_set=None, name=None, **kwargs):
         r"""Reconstruct this :class:`.WithGeometryBase` .
 
         :kwarg mesh: the new :func:`~.Mesh` (defaults to same mesh)
+        :kwarg element: the new :class:`finat.ufl.FiniteElement` (defaults to same element)
+        :kwarg boundary_set: boundary subdomain labels defining a new
+                             :func:`~.RestrictedFunctionSpace` (defaults to same boundary_set)
         :kwarg name: the new name (defaults to None)
```

Comment:
It seems that we can reconstruct named spaces without too much worry, as the new name defaults to `None`